src/script/cpp_api/s_base.cpp

```diff
@@ -489,6 +489,10 @@ void ScriptApiBase::pushPlayerHPChangeReason(lua_State *L, const PlayerHPChangeR
 		lua_pushstring(L, reason.node.c_str());
 		lua_setfield(L, -2, "node");
 	}
+	if (reason.node_pos_set) {
```

Comment:
In which case is `node_pos` not specified when `!node.empty()` ? I am asking this because intuitively, this flag is not needed and the code block can be combined with the one above.

Reply:
Yeah the flag isn't needed, I missed that. I'll do another commit removing the flag.